### ISSUE TITLE
How does this role relate to the acme_certificate module?

### ISSUE THREAD

**Issue by colans** [geerlingguy/ansible-role-certbot#87]
While searching for a solution, I came came across both the [acme_certificate](https://docs.ansible.com/ansible/latest/modules/acme_certificate_module.html) module and your role here.

I had assumed that your role would be using that module (i.e. it would simply be providing a nicer interface, defaults, etc.), but it appears as though it's not, and I can't find any reference to you using it.

I'm assuming that you either didn't know about it, it didn't exist yet, or you have some other reason for not including it.  In any case, would you kindly provide some information about this on the project page here?  A sentence/paragraph or so explaining the use cases would be fantastic.  For example, would it make more sense to use that module now because it's native?

I'm trying to decide which way to go, and any insight you have would be helpful.

**Comment by nick-brady** [comment]
Disclaimer - I'm not an expert in this. I typically stick much closer to the development side.

There are many different ways to get certs from a CA. certbot (what this repo uses) is just one of the ways which uses letsencrypt as a certificate authority. acme_certificate is more generic and if you can't use letsencrypt then it might be a good tool to check out for http-01, dns-01 and tls-alpn-01 challenges.

Certbot is a great way to manage certs from letsencrypt, so if you're needs are fairly standard this is a good choice. It's a higher level package and will also handle auto renewal (assuming you enable it).

Jeff Geerling maintains a large number of repositories for the community. I think this question is more personal than repository specific and might be better placed on stackoverflow or serverfault for a helpful answer.

**Comment by colans** [comment]
Thanks for the feedback, @nick-brady .  Are you saying that it's not possible to do renewals with `acme_certificate`?

I found https://github.com/felixfontein/acme-certificate, which looks like a good wrapper for it (if renewals can be done).

Otherwise, I'll probably stick with this `certbot` one.

**Comment by nick-brady** [comment]
no, it is possible to do renewals with `acme_certificate`. You could do something similar with letsencrypt https://docs.ansible.com/ansible/2.5/modules/letsencrypt_module.html . certbot just handles a lot of this for you, specifically for letsencrypt.

Personally I think using certbot to help manage certificates is simpler, but that's up to you. certbot is an application that handles the verification process for with the certificate authority, which is very handy. With acme_certificate you'll essentially be doing that yourself. It also automatically can set up automatic renewals which is very nice.

**Comment by colans** [comment]
Thanks for the help.  I went with the role here, but did pose this question at https://devops.stackexchange.com/questions/8722/what-are-some-reasons-for-using-an-ansible-role-for-certbot-vs-ansibles-native .  Please feel free to repost your answer there!
